### ISSUE TITLE
Why batch normalization layers's output depend on batch_size while testing

### ISSUE THREAD

**Issue by AkatsukiTse** [fchollet/keras#3078]
I have a model using batch normalization layer.what confuse me is:
`print(model.predict(x)) #return [0.9, 0.8, 0.7]`
`print(model.predict(x[1:])) #return [0.83, 0.73]`
I know the layer normalize the input and with increasing the batch_size, the difference wil be smaller.
But according to  [Batch Normalization: Accelerating Deep Network Training by Reducing
Internal Covariate Shift](http://jmlr.org/proceedings/papers/v37/ioffe15.pdf).
In test, the batch normalization use running_mean and running_variance which are caculated in train.
So why batch normalization layer normalize the input depending on the batch_size and other inputs in the batch?
Does it use the batch_mean and batch_variance to normalize the input?

**Comment by maoshengyang** [comment]
Hi~ any idea on solving this issue please? Thanks a lot!

**Comment by VishalVaddina** [comment]
**Tried and Tested**
@maoshengyang 
I'm guessing you might've already solved this by now, but in-case you're still wondering: 
You should explicitly mention **training=False** during validation and testing. Only then it would use population statistics to make the predictions, and not the batch statistics(mean and variance).

After you've mentioned training=False, no matter what batch_size of files you send for testing, predictions (logits) for respective individual test-data-points will always be same.

**Comment by henie5** [comment]
Hi, I wonder where can I set the option 'training=False ' in Keras?
